### DISCUSSION TITLE
Zigbee2mqtt and TubesZB

### DISCUSSION THREAD

**Discussion by elminster13** [tube0013/tube_gateways#107] [Q&A] [answered]
I have recently bought the TubesZB POE device and connected it to my network. I plan on using Z2M and have brought up a HAOS VM with MQTT and Z2M addons installed but I am getting a 502 Bad Gateway on start up.

As part of trying to figure out what is going on I have a question about the TubesZB firmware, I'm a bit confused as to whether the stock firmware should work with Z2M or not? Do I need to flash before it will work?

If I do need to flash, I read through the instructions, and they tell me to click the prepare for flash button but that doesn't exist on the stock firmware and when I do try and do an OTA flash it errors with something about the magic something not being available (I'm not in front of my machine at the moment but ill edit that later).

So I'm a bit lost honestly! Do I need to flash and either way should I expect to be able to do it using OTA?

**Comment by tube0013** [comment, marked as answer]
The 502 error is outside the scope here and I believe usually related to a z2m config error.

Network coordinators have 2 modules a zigbee module and an esp32 for the Ethernet/serial connection.

The Coordinators come flashed with the current z-stack release on the zigbee module which is fully compatible with zigbee2mqtt and zha/HomeAssistant. 

The web OTA is for the esp32 fw - which is esphome. 

The prepare for firmware update button on the web interface is to put the zigbee module into bootloader mode so an external application can flash the fw.

**Reply by elminster13** [reply]
Okay thank you - that helps me. I can focus my efforts on the Z2M/ HA side to get things working now knowing that the unit I purchased should work with Z2M out of the box and without any flashing.

**Reply by tube0013** [reply]
Check the add-on logs to get a hint as to what z2m is doing
